crates/pera-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1047,43 +1049,47 @@ impl AuthorityPerEpochStore {
         })
     }
 
-    pub fn get_encrypted_decryption_key_shares(&self) -> PeraResult<Vec<Vec<u8>>> {
+    pub fn get_encryption_of_decryption_key_shares(
+        &self,
+    ) -> DwalletMPCResult<HashMap<u8, Vec<EncryptionOfNetworkDecryptionKeyShares>>> {
+        // Todo (#396): Read the decryption key share from the network DKG output
         if self.epoch() == FIRST_EPOCH_ID {
-            return Err(PeraError::Unknown(
-                "First epoch does not have decryption key shares, need to run network DKG"
-                    .to_string(),
-            ));
+            return Err(DwalletMPCError::WrongEpoch(self.epoch()));
```

Comment:
This is not true, as there can be a scenario where the chain's secret has already been generated also in the first epoch. Please add a todo here.